### DISCUSSION TITLE
MacOS / Vagrant support?

### DISCUSSION THREAD

**Discussion by michalfita** [michalfita/packer-plugin-cross#22] [Maintenance]
@dbast are you Mac user? I'm thinking about removing this Vagrant step from CI. I'm nowhere near to try to figure this out as this is run on MacOS runner while I usually steer clear from any Macs.

**Comment by dbast** [comment]
Vagrant has nothing to do with macOS. It only runs on the macos-12 Github Action Runner, because this is the only runner having vagrant (with VirtualBox backend) pre-installed. A linux runner with vagrant (+ compatible backend like VirtualBox) would also be able to run the vagrant test, but such a linux runner does not exist. This repo describes what the different Github Action Runners have pre-installed https://github.com/actions/runner-images ... vagrant on macos-12 was supper reliable in the past, but started to break some time ago when the Vagrant + VirtualBox versions have been upgraded as also discussed here https://github.com/orgs/community/discussions/73013 ... there is nothing on our side to debug or do about it 

Testing the Vagrant file via Github Workflow was added in the past to prevent the Vagrantfile from rotting = being broken, when one wants to use it as described in the README. Though we can:
* Disable the vagrant tests in the workflow, but still keep the Vagrantfile
* Make the vagrant job ignore errors (shows then a potential missleading green tick in PRs)
* Extract the vagrant test into its own workflow file with `workflow_dispatch` (=manually triggered not via PRs / brnach pushes)

Can look into option 3 in the next days.

**Comment by michalfita** [comment]
Right.
I really "love" GitHub Actions... so many devils in details.
Looks like we should have self-hosted runner with Ubuntu + VirtualBox + Vagrant to run this by different means. I don't have resources to spare to run one.
